ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -109,6 +109,15 @@ def agent_id_to_index(self) -> Dict[AgentId, int]:
     def contains_agent(self, agent_id: AgentId) -> bool:
         return agent_id in self.agent_id_to_index
 
+    def get_index(self, agent_id: AgentId) -> int:
+        if not self.contains_agent(agent_id):
+            raise IndexError(
+                "get_index failed. agent_id {} is not present in the BatchedStepResult".format(
+                    agent_id
+                )
+            )
+        return self._agent_id_to_index[agent_id]  # type: ignore
```

Comment:
How about not calling `contains_agent` and just do `self.agent_id_to_index.get(agent_id, -1)`?

Reply:
No problem. I was following suit with get_agent_step_result below.